Add PresenceDB

vascYT opened the pull request:
## Screenshots 🖼️
[Home](https://files.vasc.dev/grUm5wZVk.png)
[Search](https://files.vasc.dev/KlCoEzejG.png)
[FAQ](https://files.vasc.dev/S7x8cD6gz.png)
[Activity](https://files.vasc.dev/rD7hJX260.png)
[User](https://files.vasc.dev/UmcpGXSTI.png)
[Settings](https://files.vasc.dev/0FQHCyAvP.png)

Commit:
Added PresenceDB v1.0.0

Commit:
Added translation for "view page" button

Commit:
Updated tags

Commit:
Updated images to imgur

Commit:
Fixed plex extension

I accidentally edited the plex extension. I reverted the changes. Sorry

Commit:
Fixed issues from codacy check

Commit:
really fixed the codacy check issues

vascYT committed:
Merge branch 'PreMiD:master' into master

vascYT committed:
Merge branch 'PreMiD:master' into master

vascYT committed:
Merge branch 'PreMiD:master' into master

vascYT committed:
Merge branch 'PreMiD:master' into master

vascYT committed:
Merge branch 'PreMiD:master' into master

vascYT committed:
Removed startTimestamp, used ids instead of selector

vascYT committed:
Merge branch 'PreMiD:master' into master

vascYT committed:
Added comma
(really important)

vascYT committed:
Updated thumbnail

vascYT committed:
Removed comma

Bas950 commented on websites/P/PresenceDB/dist/metadata.json:
```diff
@@ -0,0 +1,21 @@
+    "color": "#fff",
+    "tags": ["presencedb", "database"],
+    "category": "other",
+    "regExp": "(www|preview)[.]presencedb[.]com[/]([a-z]*)",
+    "readLogs": false
```
```suggestion
    "regExp": "(www|preview)[.]presencedb[.]com[/]([a-z]*)"
```

Bas950 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,90 @@
+const presence = new Presence({
+    clientId: "840759396103749633"
+}),
+{ language } = window.navigator;
+
```
use presence settings instead

Bas950 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,90 @@
+                url: window.location.href
+            }
+        ]
+    }; /*Optionally you can set a largeImageKey here and change the rest as variable subproperties, for example presenceSata.type = "blahblah"; type examples: details, state, etc.*/
+
```
remove comment

Bas950 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,90 @@
+        largeImageKey: "logo",
+        buttons: [
+            {
+                label: getTranslation("view"),
+                url: window.location.href
```
make sure this isnt there on the homepage, since buttons to the homepage are not allowed.

Bas950 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,90 @@
+    } else if (window.location.pathname === "/settings") presenceData.details = getTranslation("settings");
+
+    if (presenceData.details === null) {
+        presence.setTrayTitle(); //Clears the tray title for mac users
+        presence.setActivity(); /*Update the presence with no data, therefore clearing it and making the large image the Discord Application icon, and the text the Discord Application name*/
+    } else presence.setActivity(presenceData); //Update the presence with all the values from the presenceData object
```
remove comments

vascYT committed:
✅ Resolved requested changes

- Added language to settings
- Removed readLogs from metadata
- Buttons now only show on activity and user pages
- Removed all comments

code-factor committed:
[CodeFactor] Apply fixes to commit f0a28b1

vascYT committed:
Resolved CodeFactor issues

vascYT committed:
Updated formatting

Bas950 commented on websites/P/PresenceDB/dist/metadata.json:
```diff
@@ -0,0 +1,29 @@
+  },
+  "url": ["www.presencedb.com", "preview.presencedb.com"],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/tXfdJDT.png",
```
logo must be at least 512x512

Bas950 commented on websites/P/PresenceDB/dist/metadata.json:
```diff
@@ -0,0 +1,29 @@
+  "color": "#fff",
+  "tags": ["presencedb", "database"],
+  "category": "other",
+  "regExp": "(www|preview)[.]presencedb[.]com[/]([a-z]*)",
```
```suggestion
  "regExp": "(www|preview)[.]presencedb[.]com[/]",
```
You shouldn't need that (just double check that it still works)

vascYT committed:
✅ Updated requested changes

- Logo is now 512x512px
- Changed regex expression

vascYT committed:
Added startTimestamp

vascYT committed:
Added startTimestamp

vascYT commented:
Why is the Schema Validation Check still waiting :/

vascYT committed:
Merge branch 'master' of https://github.com/vascYT/Presences

SlowLife1661 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,103 @@
+  } else if (window.location.pathname === "/settings")
+    presenceData.details = await getTranslation("settings");
+
+  if (presenceData.details === null) {
```
```suggestion
  if (!presenceData.details) {
```

SlowLife1661 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,103 @@
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: new Date().getTime()
```
Since it's in a loop here so... the time will always be the current time, and you need to `/ 1000` it

vascYT replied:
Makes sense, gonna change that later.

vascYT replied:
But what's really weird is that is actually worked that way...

Bas950 replied:
it works because the app doesn't update the presence if the timestamp only changed 1ms
But these are still unneeded updates to the app.

vascYT committed:
Update websites/P/PresenceDB/presence.ts

Co-authored-by: Slowlife <slowlife1165@daum.net>

vascYT committed:
Update startTimestamp

vascYT committed:
Fix language icon in settings

Bas950 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,103 @@
+const presence = new Presence({
+  clientId: "840759396103749633"
+});
```
```suggestion
}),
  browsingStamp = Math.floor(Date.now() / 1000);
```

vascYT replied:
Great idea.

Bas950 commented on websites/P/PresenceDB/presence.ts:
```diff
@@ -0,0 +1,103 @@
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: Math.floor(new Date().getTime() / 1000)
```
```suggestion
    startTimestamp: browsingStamp 
```

vascYT committed:
Update websites/P/PresenceDB/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

vascYT committed:
Update websites/P/PresenceDB/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>
